File: dart/behavioral/strategy/guru_example/bin/context.dart
import 'strategy.dart';

class Context {
  Strategy _strategy;

  Context(Strategy s) {
    _strategy = s;
  }

  void setStrategy(Strategy s) {
    _strategy = s;
  }

  List<String> bloc() {
    print('Context bloc');
    return _strategy.doAlgorithm(['a', 'b', 'c', 'd', 'e']);
  }
}

File: typescript/behavioral/strategy/guru_example/context.ts
import {Strategy} from "./strategy";

export class Context {
  private strategy: Strategy

  constructor(strategy: Strategy) {
    this.strategy = strategy
  }

  public setStrategy(strategy: Strategy): void {
    this.strategy = strategy
  }

  public bloc(): string[] {
    console.log('Context bloc')
    return this.strategy.doAlgorithm(['a', 'b' ,'c', 'd', 'e', 'f', 'g'])
  }
}
